zap/src/main/java/org/zaproxy/zap/extension/authentication/ExtensionAuthentication.java

```diff
@@ -106,8 +113,9 @@ public String getUIName() {
     @Override
     public void hook(ExtensionHook extensionHook) {
         super.hook(extensionHook);
+        this.extensionHook = extensionHook;
```

Comment:
This will not work, once the hook phase is done any additions to the `extensionHook` have no effect.

Reply:
I am not entirely sure I understand what do you mean. Adding a new `AbstractAuthenticationMethodOptionsPanel` to the UI from a ZAP extension through a dynamically added `AuthenticationMethodType` was working.

Reply:
The class `AbstractAuthenticationMethodOptionsPanel` is unrelated to the `ExtensionHook` (that's part of the method itself), change your method to use the `extensionHook`. Start ZAP, install your add-on, and see if it works properly.